TumbleBitSetup.Tests/ProtocolTests.cs

```diff
@@ -399,12 +395,9 @@ public bool _ShortN(BigInteger Exp, int shortKeySize, int longKeySize)
             // A case where "shortKeySize"-bits N is used for proving and "longKeySize"-bits is needed for verifying.
             var keyPair = new RsaKey(Exp, shortKeySize);
 
-            var privKey = keyPair._privKey;
-            var pubKey = new RsaPubKey(keyPair);
-
-            byte[][] signature = PermutationTest.Proving(privKey.P, privKey.Q, privKey.PublicExponent, alpha, ps);
+            var signature = keyPair.CreatePermutationTest(setup);
             // passing "longKeySize" as the keySize.
-            return PermutationTest.Verifying(pubKey, signature, alpha, longKeySize, ps);
+            return keyPair.PublicKey.VerifyPermutationTestProof(signature, setup);
```

Comment:
Given the new API, the key length is derived from the public key ([here](https://github.com/osagga/TumbleBitSetup/pull/23/files#diff-02435bfa7e267216b41d7e93aaf9ffd2R42)) and also the same for the proving function of PoupardStern ([here](https://github.com/osagga/TumbleBitSetup/pull/23/files#diff-2cee518c1ad702c7be8b755d61df42c0R84)). I think the fix would be to make PermutationTestSetup take a keyLength as a parameter along with alpha, ps and k. And the same for PoupardSternSetup.